### ISSUE TITLE
Use different color for tertiary?

### ISSUE THREAD

**Issue by RicoElectrico** [gravitystorm/openstreetmap-carto#1974]
I think new white color of tertiary roads is misleading, kind of breaking the mapper feedback for novices. It should be clear that tertiary is meant for roads that are somehow more important than residential, and the thickness gradation isn't really that easy to see.
Another issue is that in some areas [like here](http://www.openstreetmap.org/#map=13/50.8376/16.6445), tertiary is overused as a default. When it was yellow, it was very easy to see and it would catch attention. I've also seen tertiary abuse in Africa or Middle East cities, but I don't remember where exactly.

**Comment by polarbearing** [comment]
Could you specify what you mean with "overuse"? In your example, there are some main roads going through the villages, so tertiary appears fine to me (subject to the country-specific tertiary definition and ground survey of priorities). 

There are certainly pros and cons of white wider tertiary, but they have been discussed here at length, and a decision made in favour. For me, the width works fine.

**Comment by RicoElectrico** [comment]
My reasoning is that in this example tertiary is effectively the lowest tier besides driveways. If these were larger villages (that is with residential streets branching off), putting tertiary everywhere would be fine.
And do you see something? Judging relative width is not feasible when there are no thinner roads around. That's my point.
A slight yellow tint would be OK. Or some other minor variation, it doesn't have to be prominent.

It's always like that, no matter how big discussion is, if it can't be deployed as a public beta, there always will be something that people find not right. So because I guess public preview is not technically feasible due to resources needed, no discussion should be deemed final.

**Comment by math1985** [comment]
There has been a public beta. It was announced [here](https://lists.openstreetmap.org/pipermail/talk/2015-August/073815.html).

**Comment by Teiron** [comment]
I would like to express my support to this idea. Tertiary roads don't stand out enough.

But more importantly I keep hearnig about people feeling that road color change wasn't announced properly, despite great lot of talk at mailing lists, forums and here. I think there should be a easliy visible button on mainpage, which should link to "beta" version of map. This map may not be hosted permamently but maybe only one or two weeks before realese of new version of style. Users should be able to easliy give feedback somewhere where the developers will read it. Also button should be greyed out when beta is inactive. 

This should quell some critcism, while enhancing cooperation within comunity.

**Comment by kocio-pl** [comment]
I think tertiary are just acceptable now, but I don't know how exactly they should be made better ([the discussion about it](https://github.com/gravitystorm/openstreetmap-carto/pull/1736#issuecomment-130019346) brought us no working solution). So please try to focus on propositions to test rather than complain about the situation. The problem is already known, but we have no idea how to solve it and only constructive criticism can help us. I'm happy to make test renderings if someone has the idea.

Announcing big changes may be a good idea, however they are rare (like road redesign or making the buildings pale) and most of the changes are rather gradual.

**Comment by zibik** [comment]
My propose for tertiary colour (#feffdf):
![tertiary-feffdf-kadr](https://cloud.githubusercontent.com/assets/702288/11187065/25f729f2-8c85-11e5-8ea8-3cda20138897.png)
![tertuary-feffe3](https://cloud.githubusercontent.com/assets/702288/11187823/4212599c-8c88-11e5-94da-d163e92c5310.png)

**Comment by math1985** [comment]
Looks fine to me, and in any case better than the 

Personally I'm am still not sure what to think about the white tertiary roads. On the one hand, it makes the map look calmer because there is one colour less, and in most situations, width is still enough to recognize the tertiaries. In the other hand, there are also situations where it is possible to confuse two nearby residential roads with one tertiary, such as [here](http://www.openstreetmap.org/#map=17/52.08260/5.11806) the Arthur van Schendelstraatr on z14 and z15.

**Comment by polarbearing** [comment]
The proposal is not bad in the road hierarchy, however it is quite close to the educational yellow. The latter could be solved if we migrate towards boundaries for the amenities as in #1624.

**Comment by kocio-pl** [comment]
I'm not quite sure about the proposition visibility, however the similarity to "educational yellow" would not be that significant, because tertiary road and educational/hospital area probably never overlap and their shapes are enough to recognize them easily. But what about [planned farmland color change](https://github.com/gravitystorm/openstreetmap-carto/pull/1701)?

![z9e304ye](https://cloud.githubusercontent.com/assets/5439713/11202600/0e8cb6a0-8cea-11e5-9ff8-0a45320aaed1.png)

I suggest opening this image alone, since it is probably scaled to fit the column. My code branch for it is located [here](https://github.com/kocio-pl/openstreetmap-carto/tree/tertiary-zibi).

**Comment by donaldrnoble** [comment]
One solution might be to introduce a deeper purple colour for motorways, and then shift the "new" colours down one class of road. This would, IMO, overcome a few issues (from a UK perspective possibly).
* Motorways would be closer to blue, and might stand out a bit more (both complaints I have seen elsewhere), without being too close to water coloured. 
* Primary, Secondary, & Tertiary roads would be reddish-orange, orange, and yellow respectively - which is close to Ordnance Survey maps. However the "new" OSM colours have the advantage of being a more continuous intensity spectrum against importance, which I really like.
* It also stays away from the issues highlighted above, of being too close to farmland or education colours.

I appreciate that purple might cause conflicts with borders, but these are generally rendered in light pastel shades, so a darker purple motorway with casings could look quite different.

**Comment by kocio-pl** [comment]
Are we still gonna keep this proposal or consider @donaldrnoble proposition? Original @zibik  proposal would be trivial to deploy, dropping would be even easier (just closing this issue), but @donaldrnoble issue needs someone willing to get into details (hex color values) and test it.

**Comment by mboeringa** [comment]
> Are we still gonna keep this proposal or consider @donaldrnoble proposition? Original @zibik proposal would be trivial to deploy, dropping would be even easier (just closing this issue), but @donaldrnoble issue needs someone willing to get into details (hex color values) and test it.

I think the recolouring of only tertiary per the proposal here is the best option. It is minor change that may help some and doesn't really hurt anything.

We've had plenty of discussion about alternative coloring schemes but no real final consensus in the https://github.com/gravitystorm/openstreetmap-carto/pull/1736 thread. Now @matkoniecz's new color scheme is finally there, I think it has grown upon many, also considering no real complaints here..., why would we again make a big switch now?

**Comment by aceman444** [comment]
Yes, please make tertiary a bit different from the white of other roads.

**Comment by gravitystorm** [comment]
Although this was marked as closed after #2078 was merged, there was a problem with this PR that meant it was reverted in 4fd9a5dd0ebec7c55d0df085c0e1e5ed7ebb7994

**Comment by kocio-pl** [comment]
Now we have two different propositions. As I said [here](https://github.com/gravitystorm/openstreetmap-carto/pull/2219#issuecomment-231616086): if we decide to make tertiary roads yellow, do we want to manually add color for them (my solution) or use the script and change also all existing road colors (@jdhoek's solution)? I like the first option more, but I don't reject the second one.

Manual option is less intrusive (+) and a drawback is that tertiary color is close to secondary color (-), while scripted option makes the colors differences more uniform and the difference between tertiary and secondary is bigger (+), but we make global color changes (-), so we should test if we like new colors for the roads.

**Comment by jdhoek** [comment]
In terms of maintaining the balance of the road classes palette, using the script seems preferable. With #2220 in place it should be possible to tweak the various options more easily without having to know Python.

I expect the PR to be finished some time this week — I am going to add a few fixes contributed by @pnorman .

**Comment by kocio-pl** [comment]
I like your refactoring very much, especially splitting script from the color configuration and documenting things! 

If we want more balanced road colors, it means testing worldwide and discussing it, because the scope will be much broader than just tertiary.

**Comment by pnorman** [comment]
With the refactoring merged we're unblocked and can now make changes by modifying the script.

**Comment by kocio-pl** [comment]
I made a new version of #2085 after refactoring, but with more colors and sizes for tertiary roads - see #2228. This is current "tertiary-colors-only" variant, of course making a new PR version of https://github.com/gravitystorm/openstreetmap-carto/pull/2219 ("uniform-road-colors") is another possible way.

**Comment by jdhoek** [comment]
If anyone wants to experiment with the road colours via `road-colors.yaml`, change the values in that file and run:

```
./scripts/generate_road_colours.py > road-colors-generated.mss
./scripts/generate_shields.py
```

To include tertiary roads, add `- tertiary` to the list of roads.

Not all values are possible; `generate_road_colours.py` will throw an error if you pick values that cannot be converted to RGB.

**Comment by nebulon42** [comment]
> Not all values are possible; generate_road_colours.py will throw an error if you pick values that cannot be converted to RGB.

That's why I recommend to switch to [HUSL](http://husl-colors.org). Carto supports this since 0.16.

**Comment by jdhoek** [comment]
HUSL is certainly interesting, nice project. However, the trade off for the full range of the input coordinates at any combination is a distortion in the chroma values for a generated sequence of colours; the purpose of the script is to generate a naturally progressing sequence of colours. Colours are limited to what RGB supports regardless of colour definition used.

**Comment by nebulon42** [comment]
From a designer's perspective this trade-off is worth taking if I can avoid the trial-and-error process of finding a color that has a representation in RGB. At any rate it is better than using HSL, which is often the most popular choice. You also should not forget that as soon as you use functions like `lighten` and `darken` in Carto you are doing the operations in the HSL color space unless you use perceptual colors, then Carto will do the calculations in HUSL.

**Comment by pnorman** [comment]
For the colour script where it's generating a sequence, I recommend Lch over HSUL. We're not inputting in RGB or HSL and we're not using functions like lighten or darken.

**Comment by math1985** [comment]
Closing this per https://github.com/gravitystorm/openstreetmap-carto/pull/2228#issuecomment-237345032.

Thanks everybody for the work put in investigating this!
